frontend/public/features.ts

```diff
@@ -85,27 +93,30 @@ const detectOpenShift = dispatch => coFetchJSON(openshiftPath)
       : handleError(err, FLAGS.OPENSHIFT, dispatch, detectOpenShift)
   );
 
-const detectCanListNS = dispatch => {
-  const req = {
-    spec: {
-      resourceAttributes: {
-        resource: 'namespaces',
-        verb: 'list',
-      },
-    },
-  };
-  k8sCreate(SelfSubjectAccessReviewModel, req)
-    .then(
-      res => setFlag(dispatch, FLAGS.CAN_LIST_NS, res.status.allowed),
-      err => handleError(err, FLAGS.CAN_LIST_NS, dispatch, detectCanListNS)
-    );
-};
+const detectSSARRestrictions = _.transform([
```

Comment:
What is the benefit of using `_.transform` vs a simple `_.each`? I find this harder to follow.

Reply:
Well, I am transforming an array in to an object, so its more specific w/o having to read the details.  Since `_.each` depends on side effects, you never know what its doing without reading the details. 

My gut response is we have a bad habit of using `_.each` or `for (...)` or `[].forEach` or `$.each` far too often and leave a lot of bookkeeping noise in our code when we have a great utility belt of better idioms we can be using.  Personally I think `_.each` ought to be a last resort, if you really can't find something better.

I was actually expecting my use of `_.spread()` to be the thing to raise a question here.

Reply:
Def don't want to be pedantic, but it does seem like they have made good effort to follow the functional styles of the react community in a lot of this app. I see a good bit of `_.reduce`, `_.map`, etc. There is an `_.each` occasionally, but it does not appear to be the defacto the way it is in the old angular console.  Do we want to change that?

Reply:
OK, just observing that this took me a while to work out what it was doing. A comment explaining would help.

Reply:
Maybe this my own limitation, but if it takes 10 minutes to understand 20 lines of code, I feel like the code is too complicated. I'm just wondering if we can't simplify it. E.g. instead of transforming a map into an object, just having a function that loops over the items and call the SSAR check for each. Essentially that's what this is doing, right? (I understand you need to retry on errors.)